accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,245 @@
+This is a state.apply on the target system. The selected environment information determine the state which is executed.
+
+These collected data are stored into a report table with a status field.
+The status can be:
+
+- pending
+- failed
+- succeeded
+
+The initial status is `pending` which indicate that the verification has not yet happen.
+
+### Step two
+The second step is a process which run in a dedicated environment like a container which verify the result.
+The attestation container for that environment search regular for pending attestations of its environment type.
+It read the data from the report table, performs all the needed checks and write the outcome into a result
+table with a status of `succeeded` or `failed`.
+Depending of the status of all checks, it calculate and set the status of the report from `pending` to `succeeded` of `failed`.
+
+Writing the state result into the report table could be generalized by taking the state IDs as keys and the returned result as values.
+Putting this into a map and store it as Json into a data column.
+
+A new Environment Type would than require just a state and a container to attest and write the results.
+It should be possible that one attestation container image can attest multiple environment types.
+But it may also happen that a new environment type require a new type of container image.
+
+
+## Database Design
+
+The table for the configuration should be a one-to-one relation to `rhnServer`.
+It has a boolean for attestation is enabled and an environment column for the environment of that system.
+
+```sql
+CREATE TABLE suseServerCoCoAttestationConfig
+(
+        server_id NUMERIC NOT NULL
+                    CONSTRAINT suse_srvcocoatt_cnf_sid_pk PRIMARY KEY,
+        CONSTRAINT suse_srvcocoatt_cnf_sid_fk FOREIGN KEY (server_id) REFERENCES rhnServer (id),
+        enabled   BOOLEAN NOT NULL DEFAULT FALSE,
+        environment VARCHAR(128) NULL
+                    CONSTRAINT suse_srvcocoatt_cnf_env_ck
+                    CHECK(environment IN ('kvm_amd_epyc_milan', 'kvm_amd_epyc_genoa'))
+);
+```
+
+The table for the report should be a many-to-one relation to hold a history of the attestations.
+The `data` column is of type JSONB to hold any kind of data in machine readable form collected from the
+VM to be attested.
+As different environments perform different checks, multiple data structures are expected.
+
+The status is calculated from the different results. It should be set to `succeeded` when all
+results have the status `succeeded` as well. As soon as one result is set to `failed` the
+overall status of the report should also show `failed`.
+
+```sql
+CREATE TABLE suseServerCoCoAttestationReport
+(
+        id          NUMERIC NOT NULL
+                      CONSTRAINT suse_srvcocoatt_rep_id_pk PRIMARY KEY,
+        server_id   NUMERIC NOT NULL
+                      CONSTRAINT suse_srvcocoatt_rep_sid_fk REFERENCES rhnServer (id),
+        environment VARCHAR(128) NULL
```

Comment:
It will become an NUMERIC value with an enum implementation in java.
ENUM in DB require to create a type which might be a problem in "database as a service" in some clouds.